bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,249 @@
+        return updBroadcast();
+    }
+
+    public EtherRainStatusResponse commandStatus() throws EtherRainException, IOException {
+        if (!commandLogin()) {
+            throw new EtherRainException("could not login");
+        }
+
+        List<String> responseList = null;
+
+        responseList = sendGet("result.cgi?xs");
+
+        if (responseList == null) {
+            throw new EtherRainException("Empty Response");
+        }
+
+        Iterator<String> i = responseList.iterator();
+
+        EtherRainStatusResponse response = new EtherRainStatusResponse();
+
+        while (i.hasNext()) {
+            String line = i.next();
+
+            if (line.matches(RESPONSE_STATUS_PATTERN)) {
+                String command = line.replaceFirst(RESPONSE_STATUS_PATTERN, "$1");
+                String status = line.replaceFirst(RESPONSE_STATUS_PATTERN, "$2");
+
+                switch (command) {
+                    case "un":
+                        response.setUniqueName(status);
+                        break;
+                    case "ma":
+                        response.setMacAddress(status);
+                        break;
+                    case "ac":
+                        response.setServiceAccount(status);
+                        break;
+                    case "os":
+                        response.setOperatingStatus(EtherRainOperatingStatus.fromString(status));
+                        break;
+                    case "cs":
+                        response.setLastCommandStatus(EtherRainCommandStatus.fromString(status));
+                        break;
+                    case "rz":
+                        response.setLastCommandResult(EtherRainCommandResult.fromString(status));
+                        break;
+                    case "ri":
+                        response.setLastActiveValue(Integer.parseInt(status));
+                        break;
+                    case "rn":
+                        response.setRainSensor(Integer.parseInt(status) == 1 ? true : false);
+                        break;
+                    default:
+                        logger.debug("Unknown response: {}", command);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5, int zone6,
+            int zone7, int zone8) {
+        try {
+            if (sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5
+                    + ":" + zone6 + ":" + zone7 + ":" + zone8) == null) {
+                return false;
+            }
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandClear() {
+        try {
+            if (sendGet("/result.cgi?xr") == null) {
```

Comment:
Maybe return here:

```
? result = sendGet("...");
return result != null;
```

This will also outdate line 157

Reply:
Consider adding a helper method for this structure, I see you doing it 3 times but once without exception handling...

Reply:
I have cleaned this up. Did not go with a helper function, but did make them all consistent.